webapp/tests/karma/ts/effects/reports.effects.spec.ts

```diff
@@ -885,6 +908,68 @@ describe('Reports effects', () => {
       }]);
     }));
 
+    it('should allow selecting a different report while completing verification', fakeAsync(() => {
+      const selectedReports = [{
+        _id: 'report',
+        doc: {
+          _id: 'report',
+          _rev: 2,
+          contact: { _id: 'contact', name: 'name', parent: { _id: 'parent', type: 'clinic' } },
+        },
+      }];
+      authService.has.resolves(true);
+      store.overrideSelector(Selectors.getSelectedReports, selectedReports);
+      store.refreshState();
+
+      sinon.stub(Date, 'now').returns(1000); // using faketimers breaks fakeAsync's tick :(
+      // Getting the report from the db causes a new report to be selected
+      dbService.get.callsFake(() => {
+        actions$ = concat(actions$, of(ReportActionList.selectReport({id: 'report', silent: false})));
```

Comment:
Shoot you are right. Fixed so that we start with `report` being selected, then when we verify `report`, the dbService.get function will try to select `report1`.  Then the dbService.put call will try to re-select `report` and finally the last call to selectReport will select `report2`.